spec/models/user_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User do
+  # rubocop:disable RSpec/ExampleLength
+  it 'populates a User object' do
+    # rubocop:disable Layout/LineLength
+    dataverse_user_group = 'cn=edu:berkeley:app:calnet-spa:group-spa-ucblibdataverse,ou=campus groups,dc=berkeley,dc=edu'
+    # rubocop:enable Layout/LineLength
+
+    auth = {
+      'provider' => 'calnet',
+      'extra' => {
+        'berkeleyEduAffiliations' => 'expected affiliation',
+        'displayName' => 'expected display name',
+        'berkeleyEduAlternateID' => 'expected email',
+        'uid' => 'expected UID'
+      }
+    }
+
+    [true, false].each do |dataverse_user|
+      auth['extra']['berkeleyEduIsMemberOf'] = dataverse_user ? dataverse_user_group : ''
+      user = described_class.from_omniauth(auth)
+      expect(user).to have_attributes(
+        affiliations: 'expected affiliation',
+        display_name: 'expected display name',
+        email: 'expected email',
+        uid: 'expected UID'
+      )
+    end
+  end
+  # rubocop:enable RSpec/ExampleLength
+
+  it 'returns true if user has an ID' do
```

Comment:
should these examples reference the `#authenticated?` method in their description?

Reply:
I wrapped those two examples in a describe '#authenticated?' block - I'm assuming that's what you meant for this section.